### PR TITLE
Simple tests for base 64 decoding

### PR DESCRIPTION
**Pull request by KMariusz**
http://10.40.234.40:8080/job/MrChecker-SDLC/job/CI_mrchecker-webapi-module/65/testReport/com.capgemini.mrchecker.webapi.httpbin/Base64Test/

### CODE REVIEW

**Comment by KMariusz** [comment]
https://github.com/devonfw/devonfw-testing/issues/130

File: mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+public class Base64Test extends BasePageWebApiTest {
+	private Base64Page base64Page = new Base64Page();
+
+	@Test
+	public void base64DecodeAwesome() {
+		String validBase64Data = "SFRUUEJJTiBpcyBhd2Vzb21l";
+		String validBase64Decode = "HTTPBIN is awesome";
+		BFLogger.logInfo(MessageFormat.format("Step 1 - Sending GET query to {0} with valid base 64 data: {1}", base64Page.getEndpoint(), validBase64Data));
+		Response response = base64Page.sendGETQuery(validBase64Data);
```
Comment (przwojtkow) [inline mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java]: Please take a look on duplicated blocks of code and then wrap them as a separate private method to reuse.

**Review by przwojtkow — CHANGES_REQUESTED** [review]
CR done, please check comment.

File: mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.text.MessageFormat;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class Base64Test extends BasePageWebApiTest {
+	private Base64Page base64Page = new Base64Page();
+
+	@Test
+	public void base64DecodeAwesome() {
```
Comment (przwojtkow) [inline mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java]: Test name is still a method name, thus should be a verb. Please consider changing the name of both tests.

**Comment by przwojtkow** [comment]
Please provide new CI output.

**Comment by KMariusz** [comment]
New report
http://10.40.234.40:8080/job/MrChecker-SDLC/job/CI_mrchecker-webapi-module/74/testReport/com.capgemini.mrchecker.webapi.httpbin/Base64Test/

File: mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	@Test
+	public void validateCorrectnessOfBase64Decode_MrCheckerText() {
+		String validBase64Data = "TXIuQ2hlY2tlcg==";
+		String validBase64Decode = "Mr.Checker";
+		base64DecodeDataCheck(validBase64Data, validBase64Decode);
+	}
+
+	private void base64DecodeDataCheck(String base64Data, String expectedResult) {
```
Comment (pawel-szczesny) [inline mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java]: Please consider to use Parameters for this tests.
Example from https://www.testwithspring.com/lesson/writing-parameterized-tests-with-junit-4/:
```
@RunWith(JUnitParamsRunner.class)
@Category(UnitTest.class)
public class JUnitParamsCalculatorTest {
 
    @Test
    @Parameters({
            "0, 0, 0",
            "1, 1, 2"
    })
    public void shouldReturnCorrectSum(int first, 
                                       int second, 
                                       int expectedSum) {
        Calculator calculator = new Calculator();
        int actualSum = calculator.sum(first, second);
        assertThat(actualSum).isEqualByComparingTo(expectedSum);
    }
}
```

**Review by pawel-szczesny — CHANGES_REQUESTED** [review]

**Comment by KMariusz** [comment]
Parametrized. Report:
http://10.40.234.40:8080/job/MrChecker-SDLC/job/CI_mrchecker-webapi-module/79/testReport/com.capgemini.mrchecker.webapi.httpbin/Base64Test/

**Review by pawel-szczesny — APPROVED** [review]
Good job - approve

File: mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	})
+	public void sendGetWithBase64DataAndValidateDecodedResponse(String base64Data, String expectedResult) {
+		BFLogger.logInfo(MessageFormat.format("Step 1 - Sending GET query to {0} with valid base 64 data: {1}", base64Page.getEndpoint(), base64Data));
+		Response response = base64Page.sendGETQuery(base64Data);
+
+		BFLogger.logInfo("Step 2 - Validate response status code (should be 200): ");
+		assertThat(response.statusCode(), is(200));
+
+		BFLogger.logInfo(MessageFormat.format("Step 3 - Validate response body (should be {0}): ", expectedResult));
+		assertThat(response.body().asString(), is(expectedResult));
```
Comment (tompalka) [inline mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java]: As response will always be `String`, we could move getting decoded value to `Base64Page` class
For example
`base64Page.sendGETQuery(base64Data).getDecodedValue()`
or even simpler
`base64Page.getDecodedValue(base64Data)`

**Review by tompalka — CHANGES_REQUESTED** [review]

**Review by michalbabiarz — CHANGES_REQUESTED** [review]
This change should be integrated into: https://github.com/devonfw/devonfw-testing

File: mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java (hunk 4 above)
Comment (KMariusz) [reply mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/Base64Test.java]: Done

**Comment by KMariusz** [comment]
http://10.40.234.40:8080/job/MrChecker-SDLC/job/CI_mrchecker-webapi-module/86/testReport/com.capgemini.mrchecker.webapi.httpbin/Base64Test/

**Review by tompalka — APPROVED** [review]

**Comment by tompalka** [comment]
@przwojtkow Please review and merge if everything is ok

**Comment by pawel-szczesny** [comment]
> @przwojtkow Please review and merge if everything is ok

I hope that 2 approvals will be enough for this task so I will merge it

**Pull request merged**